src/mod/applications/mod_avmd/mod_avmd.c

```diff
@@ -632,9 +636,9 @@ static switch_bool_t avmd_callback(switch_media_bug_t *bug, void *user_data, swi
 
         case SWITCH_ABC_TYPE_CLOSE:
 
-            switch_mutex_lock(avmd_globals.mutex);
             avmd_session_close(avmd_session);
 
+            switch_mutex_lock(avmd_globals.mutex);
```

Comment:
Do we have to unlock the `avmd_sessions.mutex` that we've locked on line 618 first before locking this mutex?

Reply:
This is a different mutex, this is a avmd_global mutex not avmd_session mutex